tmt/schemas/provision/hardware.yaml

```diff
@@ -145,6 +145,32 @@ definitions:
     items:
       "$ref": "#/definitions/network"
 
+  # HW requirements: single `system` item
+  system:
+    type: object
+
+    properties:
+      vendor:
+        type: string
+
+      model:
+        type: string
+
+      numa-nodes:
+        type: string
+
+    additionalProperties: false
+
+    # enforce at least one property
+    minProperties: 1
+
+  # HW requirements: `system` block
+  systems:
+    type: array
+    items:
+      "$ref": "#/definitions/system"
```

Comment:
I don't think we want to support mutliple `system` definitions. For `network` or `disks` it makes sense as we can require a guest with multiple network cards or multiple disks. But there would always we a single `vendor` or single number of numa nodes.

Reply:
@happz, @thrix, @lukaszachy, what are your thoughts on this?

Reply:
I'd say there might be a use case for getting a machine matching one of many possible vendors. I see no problem with that. On the other hand, I would stay away from `system` being a list, to keep things readable, and it's less common than NICs or storage devices. If there's a need to pick one of the multiple allowed system vendors, there's always `and` and `or`.